src/command.rs

```diff
@@ -0,0 +1,306 @@
+            .to_str()
+            .wrap_err(DIR_ERR)?
+            .parse()?,
+    };
+
+    const VERSION: Version = Version::new(0, 1, 0);
+
+    let manifest = Manifest {
+        package: PackageManifest {
+            r#type,
+            name,
+            version: VERSION,
+            description: None,
+        },
+        dependencies: vec![],
+    };
+
+    manifest.write().await?;
+
+    PackageStore::create().await
+}
+
+/// Adds a dependency to this project
+pub async fn add(dependency: String) -> eyre::Result<()> {
+    let lower_kebab = |c: char| (c.is_lowercase() && c.is_ascii_alphabetic()) || c == '-';
+
+    let (repository, dependency) = dependency
+        .trim()
+        .split_once('/')
+        .wrap_err("Invalid dependency specification")?;
+
+    ensure!(
+        repository.chars().all(lower_kebab),
+        "Repositories must be in lower kebab case"
+    );
+
+    let (package, version) = dependency
+        .split_once('@')
+        .wrap_err_with(|| format!("Invalid dependency specification: {dependency}"))?;
+
+    let package = package
+        .parse::<PackageName>()
+        .wrap_err_with(|| format!("Invalid package name supplied: {package}"))?;
+
+    let version = version
+        .parse::<VersionReq>()
+        .wrap_err_with(|| format!("Invalid version requirement supplied: {package}"))?;
+
+    let mut manifest = Manifest::read().await?;
+
+    manifest
+        .dependencies
+        .push(Dependency::new(repository.to_owned(), package, version));
+
+    manifest.write().await
+}
+
+/// Removes a dependency from this project
+pub async fn remove(package: PackageName) -> eyre::Result<()> {
+    let mut manifest = Manifest::read().await?;
+
+    let dependency = manifest
+        .dependencies
+        .iter()
+        .find(|d| d.package == package)
+        .wrap_err(eyre::eyre!(
+            "Unable to remove unknown dependency {package:?}"
+        ))?
+        .to_owned();
+
+    manifest.dependencies.retain(|d| *d != dependency);
+
+    PackageStore::uninstall(&dependency.package).await.ok();
+
+    manifest.write().await
+}
+
+/// Packages the api and writes it to the filesystem
+pub async fn package(directory: String, dry_run: bool) -> eyre::Result<()> {
+    let package = PackageStore::release()
+        .await
+        .wrap_err("Failed to create release")?;
+
+    let path = Path::new(&directory).join(format!(
+        "{}-{}.tgz",
+        package.manifest.package.name, package.manifest.package.version
+    ));
+
+    if !dry_run {
+        std::fs::write(path, package.tgz).wrap_err("failed to write package to filesystem")?;
+    }
+
+    Ok(())
+}
+
+/// Publishs the api package to the registry
+pub async fn publish(
+    credentials: Credentials,
+    repository: String,
+    allow_dirty: bool,
+    dry_run: bool,
+) -> eyre::Result<()> {
+    if let Ok(repository) = git2::Repository::discover(Path::new(".")) {
+        let statuses = repository
+            .statuses(None)
```

Comment:
I wrote the same code yesterday and irrc this doesn't filter out ignored files - rather you need to do `Some(Default::default())`. Did you test this?

Likewise - why does this matter to buffrs? Surely for buffrs there are like 2 files that matter and who cares if the user has a bunch of other files edited - that's not buffrs' purpose.

Reply:
Tracked in #61